rpc/src/v1/types/transaction_with_hash.rs

```diff
@@ -0,0 +1,134 @@
+    #[serde(rename_all = "camelCase")]
+    AssetTransfer {
+        network_id: NetworkId,
+        burns: Vec<AssetTransferInput>,
+        inputs: Vec<AssetTransferInput>,
+        outputs: Vec<AssetTransferOutput>,
+        hash: H256,
+    },
+    #[serde(rename_all = "camelCase")]
+    AssetCompose {
+        network_id: NetworkId,
+        shard_id: ShardId,
+        metadata: String,
+        registrar: Option<PlatformAddress>,
+        inputs: Vec<AssetTransferInput>,
+        output: AssetMintOutput,
+        hash: H256,
+    },
+    #[serde(rename_all = "camelCase")]
+    AssetDecompose {
+        network_id: NetworkId,
+        input: AssetTransferInput,
+        outputs: Vec<AssetTransferOutput>,
+        hash: H256,
+    },
+    #[serde(rename_all = "camelCase")]
+    AssetUnwrapCCC {
+        network_id: NetworkId,
+        burn: AssetTransferInput,
+        hash: H256,
+    },
+}
+
+impl From<TransactionType> for TransactionWithHash {
+    fn from(from: TransactionType) -> Self {
+        let hash = from.hash();
+        match from {
+            TransactionType::AssetMint {
+                network_id,
+                shard_id,
+                metadata,
+                registrar,
+                output,
+            } => TransactionWithHash::AssetMint {
+                network_id,
+                shard_id,
+                metadata,
+                registrar: registrar.map(|registrar| PlatformAddress::new_v1(network_id, registrar)),
+                output,
+                hash,
+            },
+            TransactionType::AssetTransfer {
+                network_id,
+                burns,
+                inputs,
+                outputs,
+            } => TransactionWithHash::AssetTransfer {
+                network_id,
+                burns,
+                inputs,
+                outputs,
+                hash,
+            },
+            TransactionType::AssetCompose {
+                network_id,
+                shard_id,
+                metadata,
+                registrar,
+                inputs,
+                output,
+            } => TransactionWithHash::AssetCompose {
+                network_id,
+                shard_id,
+                metadata,
+                registrar: registrar.map(|registrar| PlatformAddress::new_v1(network_id, registrar)),
+                inputs,
+                output,
+                hash,
+            },
+            TransactionType::AssetDecompose {
+                network_id,
+                input,
+                outputs,
+            } => TransactionWithHash::AssetDecompose {
+                network_id,
+                input,
+                outputs,
+                hash,
+            },
+            TransactionType::AssetUnwrapCCC {
+                network_id,
+                burn,
+            } => TransactionWithHash::AssetUnwrapCCC {
+                network_id,
+                burn,
+                hash,
+            },
+        }
+    }
+}
```

Comment:
Would you install https://github.com/editorconfig/editorconfig-vscode?
https://github.com/CodeChain-io/codechain/blob/master/.editorconfig#L10
There mustbe new line at the end since we set `insert_final_newline`.